shipment_advice_planner_toursolver/models/toursolver_task.py

```diff
@@ -0,0 +1,495 @@
+        )
+        time_windows = []
+        if delivery_windows:
+            for window in delivery_windows:
+                time_windows.append(
+                    {
+                        "beginTime": window.float_to_time_repr(window.start),
+                        "endTime": window.float_to_time_repr(window.end),
+                    }
+                )
+        else:
+            time_windows.append(self._get_default_delivery_window())
+        return time_windows
+
+    def _get_default_delivery_window(self):
+        self.ensure_one()
+        delivery_window_model = self.env["toursolver.delivery.window"]
+        backend = self.toursolver_backend_id
+        return {
+            "beginTime": delivery_window_model.float_to_time_repr(
+                backend.delivery_window_start
+            ),
+            "endTime": delivery_window_model.float_to_time_repr(
+                backend.delivery_window_end
+            ),
+        }
+
+    def _toursolver_json_request_resources(self):
+        return [
+            self._toursolver_json_request_resource(resource)
+            for resource in self.delivery_resource_ids
+        ]
+
+    def _toursolver_json_request_resource(self, resource):
+        res = resource._get_resource_properties()
+        res.update(self._toursolver_json_request_resource_start_end_position(resource))
+        work_start_time = self.toursolver_backend_id._get_work_start_time_formatted()
+        loading_duration = self.toursolver_backend_id._get_loading_duration_formatted()
+        res.update(
+            {
+                "workStartTime": work_start_time,
+                "fixedLoadingDuration": loading_duration,
+                # can be resource prop?
+                "travelPenalty": self.toursolver_backend_id.travel_penalty,
+                "workPenalty": self.toursolver_backend_id.work_penalty,
+            }
+        )
+        return res
+
+    def _toursolver_json_request_resource_start_end_position(self, resource):
+        address = self.warehouse_id.partner_id
+        res = {
+            "startX": address.partner_longitude,
+            "startY": address.partner_latitude,
+            "endX": address.partner_longitude,
+            "endY": address.partner_latitude,
+        }
+        if resource.use_delivery_person_coordinates_as_end:
+            delivery_person_id = resource.delivery_person_id
+            res.update(
+                {
+                    "endX": delivery_person_id.partner_longitude,
+                    "endY": delivery_person_id.partner_latitude,
+                }
+            )
+        return res
+
+    def _toursolver_json_request_options(self):
+        self.ensure_one()
+        res = {
+            "vehicleCode": "deliveryIntermediateVehicle",
+            "maxOptimDuration": seconds_to_duration(
+                self.toursolver_backend_id.duration
+            ),
+            "useForbiddenTransitAreas": False,
+        }
```

Comment:
These hard coded values should come from a toursolver.option record link to the backend as for the resources.